cql3/restrictions/partition_key_restrictions.cc

```diff
@@ -164,5 +169,91 @@ bool partition_key_restrictions::key_is_in_relation() const {
 bool partition_key_restrictions::pk_restrictions_need_filtering() const {
     return !is_empty_restriction(_filtering_restrictions);
 }
+
+dht::partition_range_vector partition_key_restrictions::get_partition_key_ranges(const query_options& options) const {
+    if (_column_eq_restrictions.size() == _table_schema->partition_key_size()) {
+        return get_partition_key_ranges_from_column_eq_restrictions(options);
+    }
+    return get_partition_key_ranges_from_token_range_restrictions(options);
```

Comment:
Shouldn't we make this choice during prepare time?

Reply:
The goal is to have the plans decided on during prepare time, then we can make a single dispatch (calling the correct execute() function) with no additional checks. Of course we still have multiple plans in one concrete select_statement implementation, but at least we have a choice.